azurelinuxagent/common/osutil/ubuntu.py

```diff
@@ -60,6 +70,30 @@ def register_agent_service(self):
     def unregister_agent_service(self):
         return shellutil.run("systemctl mask walinuxagent", chk_err=False)
 
+
+class UbuntuOSUtil(Ubuntu16OSUtil):
+    def __init__(self):
+        super(UbuntuOSUtil, self).__init__()
+
+    def restart_if(self, ifname, retries=3, wait=5):
```

Comment:
I am hesitant to change this for the 16x images as well. Can we limit to 1710?

Reply:
This change is only for 17.10 and above.